substrate/client/rpc-spec-v2/src/transaction/metrics.rs

```diff
@@ -0,0 +1,146 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Metrics for recording transaction events.
+
+use std::{collections::HashSet, time::Instant};
+
+use prometheus_endpoint::{
+	exponential_buckets, linear_buckets, register, Histogram, HistogramOpts, PrometheusError,
+	Registry,
+};
+
+use super::TransactionEvent;
+
+/// RPC layer metrics for transaction pool.
+#[derive(Debug, Clone)]
+pub struct Metrics {
+	validated: Histogram,
+	in_block: Histogram,
+	finalized: Histogram,
+	dropped: Histogram,
+	invalid: Histogram,
+	error: Histogram,
+}
+
+impl Metrics {
+	/// Creates a new [`Metrics`] instance.
+	pub fn new(registry: &Registry) -> Result<Self, PrometheusError> {
```

Comment:
nit: it would help a tiny bit mentioning the time unit in the metrics descriptions, while reading the code.